examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -143,15 +143,12 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
 
     Messaging::SendFlags sendFlags;
     System::PacketBufferHandle payloadBuf;
-    char * requestData = nullptr;
+    char requestData[kMsgBufSize];
 
     uint32_t size = gPingArguments.GetEchoReqSize();
-    VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
+    VerifyOrExit(size >= kMsgBufSize, err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
 
-    requestData = static_cast<char *>(chip::Platform::MemoryAlloc(size));
-    VerifyOrExit(requestData != nullptr, err = CHIP_ERROR_NO_MEMORY);
-
-    snprintf(requestData, size, "Echo Message %" PRIu64 "\n", gPingArguments.GetEchoCount());
+    snprintf(requestData, kMsgBufSize, "Echo Message %" PRIu64 "\n", gPingArguments.GetEchoCount());
```

Comment:
This can leave `requestData` not null-terminated, right?

Reply:
Yes, not all platforms null terminate a string when snprintf is used. Is null terminator mandatory in the final payload since it already has provided payload->DataLength()

Reply:
The payload doesn't care, obviously.  But the size being given for the payload here is `size`, which is some random number checked to be larger than the actual useful data we are putting in the packet.  What's the other side going to do with this on receipt?  I was assuming it was relying on null-termination of the "actual data" because the `DataLength()` is nonsensical in this case.  But if the intent is that `DataLength()` actually have to do something with the bytes we are actually initializing to something useful, then obviously null-termination is not relevant.

Reply:
No, the other size does not care about the content, it is set just to feed API NewWithData. We don't need it after switching to API New.